src/app/weather-dashboard/weather-dashboard.component.ts

```diff
@@ -0,0 +1,28 @@
+import { Component, OnInit } from '@angular/core';
+import { WeatherService } from '../weather.service';
+
+@Component({
+  selector: 'qpac-weather-dashboard',
+  templateUrl: './weather-dashboard.component.html',
+  styleUrls: ['./weather-dashboard.component.scss']
+})
+export class WeatherDashboardComponent implements OnInit {
+  response: Object;
+
+  constructor(private weather: WeatherService) { }
+
+  ngOnInit() {
+  }
+
+  changedDepartureDate(res) {
```

Comment:
You can destruct the `res` parameter by `changedDepartureDate({ from, lang })`
and use destructured properties like `this.weather.getWeather(from, lang)`